ci-operator/step-registry/telco5g/origin/tests/telco5g-origin-tests-commands.sh

```diff
@@ -44,30 +41,236 @@ git clone -b "${CNF_BRANCH}" "${CNF_REPO}" cnf-features-deploy
+
+# Pull secrets might be required to extract url to container image of OpenShift's conformance test suite
+oc get secret pull-secret -n openshift-config -o jsonpath="{.data['\.dockerconfigjson']}" \
+    | base64 --decode > pull-secret.txt
+
+# Pull secrets are required to pull container image for openshift-tests
+oc get secret pull-secret -n openshift-config -o yaml | grep -v '^\s*namespace:\s' | oc apply -f -
+
+# Extract url to container image of OpenShift's conformance test suite aka openshift-tests
+T5_JOB_TESTS_IMAGE=$(oc adm release info --registry-config pull-secret.txt --image-for=tests "$T5_JOB_RELEASE_IMAGE")
+
+# Launch a pod from the openshift-tests image and let it idle
+# while we copy the openshift-tests binary out of the container
+oc run telco5g-tests-extractor --restart=Never --image "$T5_JOB_TESTS_IMAGE" --command=true \
+    --overrides='{ "spec": { "template": { "spec": { "imagePullSecrets": [{"name": "pull-secret"}] } } } }' \
+    -- sleep 3600
+
+oc wait pod/telco5g-tests-extractor --for=condition=Ready --timeout=15m
+
+# Extract openshift-tests binary from container
+# shellcheck disable=SC2034
+for i in $(seq 10); do # retry because copy is flaky
+    if oc cp telco5g-tests-extractor:/usr/bin/openshift-tests openshift-tests; then
+        break
+    else
+        rm -f openshift-tests # remove incompletely transferred files
+        sleep 5
+    fi
+done
+[ -e "openshift-tests" ]
+
+oc delete pods telco5g-tests-extractor
+
+chmod a+x openshift-tests
+
+# Determine list of tests
+./openshift-tests run --dry-run --provider "${TEST_PROVIDER}" "${TEST_SUITE}" > "$ARTIFACT_DIR/tests-all.txt"
+if [ -n "${TEST_SKIPS}" ]; then
+    grep -v "$TEST_SKIPS" "$ARTIFACT_DIR/tests-all.txt" > "$ARTIFACT_DIR/tests-run.txt"
+
+    if ! grep "$TEST_SKIPS" "$ARTIFACT_DIR/tests-all.txt" > "$ARTIFACT_DIR/tests-skip.txt"; then
+      echo >&2 "ERROR: No tests were found matching the TEST_SKIPS regex:"
+      echo >&2 "$TEST_SKIPS"
+      exit 1
+    fi
+
+    TEST_FILE=tests-run.txt
+else
+    TEST_FILE=tests-all.txt
+fi
+
+IFS=- read -r CLUSTER_NAME _ <<< "$(cat "${SHARED_DIR}/cluster_name")"
+
+cat << EOF > run-openshift-tests.yml
+---
+- name: Run OpenShift's conformance test suite
+  hosts: hypervisor
+  gather_facts: false
+  any_errors_fatal: true
+
+  tasks:
+  - name: Copy OpenShift's conformance test suite through bastion host to installer, run it and retrieve results
+    block:
+    - name: Create script to run openshift-tests binary on bastion host
+      ansible.builtin.copy:
+        content: |
+          #!/bin/bash
+
+          set -o nounset
+          set -o errexit
+          set -o pipefail
+
+          cd /home/cloud-user/
+
+          export KUBECONFIG=kubeconfig
+
+          ./openshift-tests run "${TEST_SUITE}" \\
+            --provider '${TEST_PROVIDER}' \\
+            -o e2e.log --junit-dir junit --file tests.txt \\
+            > openshift-tests.log
```

Comment:
This file worth to save to artifacts for reproducing

Reply:
All those files will be stored in `openshift-tests.tar.gz` later.

Reply:
Ah, `run-openshift-tests.sh` you mean...